src/ThreadPool.zig

```diff
@@ -113,14 +103,18 @@ fn runWorker(self: *ThreadPool) void {
             continue;
         }
 
-        if (!self.is_running) {
+        if (self.is_running) {
```

Comment:
Is it possible to shutdown the thread pool before all tasks are completed?  If so, wouldn't this check need to be done before executing the next task above?

Reply:
The usage pattern in the self-hosted compiler will never call shutdown with running jobs. Even if it did, it would still be valid to have the code structured this way. I'm not concerned about this. I just want to figure out that damn deadlock on the drone CI.

Reply:
Ok, in that case we may want to document this since I think it would be unexpected.  Maybe something like this?
```
/// shutdown will cause the thread pool to stop, but only after all tasks have been completed
```

Reply:
Looking at the latest version, it looks like there isn't even a `shutdown` function anymore, so this comment may no longer apply.